go/interpreter/lfvm/converter_fuzz_test.go

```diff
@@ -0,0 +1,401 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package lfvm
+
+import (
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca/vm"
+)
+
+func FuzzLfvmConverter(f *testing.F) {
+
+	// Add empty code
+	f.Add([]byte{})
+
+	f.Fuzz(func(t *testing.T, toscaCode []byte) {
+
+		// EIP-170 stablish maximum code size to 0x6000 bytes, ~22 KB
+		// (see https://eips.ethereum.org/EIPS/eip-170)
+		// EIP-3860 stablish maximum init code size to 49_152 bytes
+		// (see https://eips.ethereum.org/EIPS/eip-3860)
+		// Before EIP-3860, any size was allowed, but in the Fantom
+		// network anything larger than 2^16 was observed, which is
+		// also the limit for the conversion code (due to a 16-bit PC
+		// counter representation in the PC instruction). Thus, we test
+		// the code up to this level.
+		maxCodeSize := 1<<16 - 1
+		if len(toscaCode) > maxCodeSize {
+			t.Skip()
+		}
+
+		type pair struct {
+			originalPos, lfvmPos int
+		}
+		var pairs []pair
+		lfvmCode := convertWithObserver(toscaCode, ConversionConfig{}, func(evm, lfvm int) {
+			pairs = append(pairs, pair{evm, lfvm})
+		})
+
+		// Check that all operations are mapped to matching operations.
+		for _, p := range pairs {
+
+			toscaOpCode := vm.OpCode(toscaCode[p.originalPos])
+			lfvmOpCode := lfvmCode[p.lfvmPos].opcode
+
+			if !vm.IsValid(toscaOpCode) && lfvmOpCode != INVALID {
+				t.Errorf("Expected INVALID, got %v", lfvmOpCode.String())
+			}
+
+			if vm.IsValid(toscaOpCode) {
+				if got, want := toscaToLfvmOpcode(toscaOpCode), lfvmOpCode; got != want {
+					t.Errorf("Expected %v, got %v", want, got)
+				}
+			}
+		}
+
+		// Check that the position of JUMPDEST ops are preserved.
+		for _, p := range pairs {
+			if vm.OpCode(toscaCode[p.originalPos]) == vm.JUMPDEST {
+				if p.originalPos != p.lfvmPos {
+					t.Errorf("Expected JUMPDEST at %d, got %d", p.originalPos, p.lfvmPos)
+				}
+			}
+		}
+	})
+}
+
+func TestFuzzerTooling_OpCodeConversionIsComplete(t *testing.T) {
+	// Check that all Tosca opcodes are mapped to LFVM opcodes.
+	for i := 0; i <= 0xFF; i++ {
+		op := vm.OpCode(i)
+		if !vm.IsValid(op) {
+			continue
+		}
+		if toscaToLfvmOpcode(op) == INVALID {
+			t.Errorf("Opcode %v is not mapped to LFVM", op)
+		}
+	}
+}
+
+func toscaToLfvmOpcode(op vm.OpCode) OpCode {
```

Comment:
maybe this function and its mirror should be in a separate file and tested. possibly another PR

Reply:
Let's wait with this until we have settled the question on whether and how we are cleaning up the op code definition.

Reply:
I make a note on that: #723